usr/src/uts/common/fs/zfs/sys/dsl_pool.h

```diff
@@ -103,6 +104,8 @@ typedef struct dsl_pool {
 	kcondvar_t dp_spaceavail_cv;
 	uint64_t dp_dirty_pertxg[TXG_SIZE];
 	uint64_t dp_dirty_total;
+	uint64_t dp_long_free_dirty_pertxg[TXG_SIZE];
+	uint64_t dp_long_freeing_total;
```

Comment:
What I was thinking is using dp_long_freeing_total to track total space we're freeing, across potentially more than 3 transaction.
This way zpool's freeing prop will line up with the file size deleted and not just 3 transactions worth of chunks that are being freed.